packages/flutter/lib/src/material/input_decorator.dart

```diff
@@ -1090,11 +1111,38 @@ class _RenderDecoration extends RenderBox {
       boxToBaseline: boxToBaseline,
       containerHeight: containerHeight,
       inputBaseline: inputBaseline,
+      outlineBaseline: outlineBaseline,
       subtextBaseline: subtextBaseline,
       subtextHeight: subtextHeight,
     );
   }
 
+  // Interpolate between three stops using textAlignVertical. This is used to
+  // calculate the outline baseline, which ignores padding for its middle but
```

Comment:
ignores padding when the alignment is middle and interpolates between the centered text box's top and the top of the content padding when the alignment is less than zero [etc...]